### PR TITLE
ispow2: change parameter to uint64_t

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by WalterBright** [comment]
Of course, this doesn't work because the bootstrap dmd uses the old scheme. Sigh.

**Comment by WalterBright** [comment]
Worked around by having a separate `ispow2` on the D side.

File: src/dmd/iasm.d [hunk 1]
```diff
@@ -4558,3 +4558,20 @@ AFTER_EMIT:
     //return asmstate.bReturnax;
     return s;
 }
+
+/**********************
+ * If c is a power of 2, return that power else -1.
+ */
+
+private int ispow2(uint c)
```
Comment (wilzbach) [inline src/dmd/iasm.d]: D shouldn't be about duplicating code :(

**Review by wilzbach — APPROVED** [review]
I assume we can get rid of this duplication with the "D-ification" of the backend soon.

File: src/dmd/iasm.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/iasm.d]: There's a lot of hackish code that the conversion to D will eliminate.

**Pull request merged**